### ISSUE TITLE
Error waiting on image with googlecompute builder 

### ISSUE THREAD

**Issue by andrewsykim** [mitchellh/packer#4685]
Version:
Packer v0.12.3

Platform:
MacOSX

Debug logs:
https://gist.github.com/andrewsykim/5abf8563dec397fcb57ff1aba703fca8

Template I used:
```json
{
  "builders": [{
    "type": "googlecompute",
    "account_file": "account.json",
    "project_id": "my-project-id",
    "source_image": "coreos-stable-1298-6-0-v20170315",
    "zone": "us-central1-a",
    "ssh_username": "core"
  }],
  "provisioners": [
    {
      "type": "shell",
      "script": "scripts/setup.sh"
    }
  ]
}
```

**Comment by rickard-von-essen** [comment]
This is a duplicate of #4118. Unfortunately that is closed and unresolved.

@mwhooker IMHO we should change how the GCE builder handles startup-scripts. It's bound to fail trying to wrap startup scripts, instead we should add a new option `wait_for_startup_script` which should default to false. If false we do nothing, if true we will wait for the change of Metadata. The startup must self take care of setting this. This should be very clear in the docs with several examples. 

The only problem with the above change is that it has no simple upgrade path and will break BC for everyone relying on waiting on startup scripts to complete.

**Comment by andrewsykim** [comment]
Thanks for the quick response! I'll see what I can do with user-data.
